### PR TITLE
Upgrade Auth0 lock.js to fix login problems

### PR DESCRIPTION
**Pull request by hiemanshu**
Also update documentation to add the new required setting.

The endpoints, and other changes made by Auth0 were not backward compatible with
the version of lock.js we were running.

### CODE REVIEW

**Review by schuyler1d — APPROVED** [review]
:+1: yei to fixing auth0 issues.

**Review by shakalee14 — APPROVED** [review]

**Comment by schuyler1d** [comment]
I think this is a good start.  When testing with a fresh auth0 account, and creating a new user by signup, on first-login the server fails with:
```
12:22:29 PM server.1              |  backend error TypeError: Cannot read pr
12:22:29 PM server.1              |  >  operty 'given_name' of undefined
12:22:29 PM server.1              |      at _callee6$ (/home/sky/sb/moveon/spoke/src/server/inde
12:22:29 PM server.1              |  >  x.js:171:21)
....
```
and the client fails with:
`Error occured while trying to proxy to: localhost:3000/login-callback?code=...` at /login-callback

I'm not completely sure why, but I believe Auth0 has changed from defaulting to send user_metadata with the profile info to needing a "Rule":
https://auth0.com/docs/metadata#use-the-auth0-apis
Though even creating a rule, I haven't been able to get our custom fields to come in.

**Comment by hiemanshu** [comment]
@schuyler1d @shakalee14 Can you take a look at this? Thanks.

File: README.md [hunk 1]
```diff
@@ -30,17 +30,25 @@ Follow up instructions located [here](https://github.com/MoveOnOrg/Spoke/blob/ma
...
 7. Run `npm run dev` to create and populate the tables.
 8. Add the login callback and logout callback URL in `.env` (default `http://localhost:3000/login-callback` and `http://localhost:3000/logout-callback`) to your Auth0 app settings under "Allowed Callback URLs" and "Allowed Logout URLs" respectively. (If you get an error when logging in later about "OIDC", go to Advanced Settings section, and then OAuth, and turn off 'OIDC Conformant')
-9. Run `npm run dev` to start the app. Wait until you see both "Node app is running ..." and "Webpack dev server is now running ..." before attempting to connect. (make sure environment variable `JOBS_SAME_PROCESS=1`)
-10. Go to `http://localhost:3000` to load the app.
-11. As long as you leave `SUPPRESS_SELF_INVITE=` blank and unset in your `.env` you should be able to invite yourself from the homepage.
+9. Add the domain name for your app to "Allowed Web Origins" in the Auth0 Settings. It should be the same as the login/logout callback URLs without the callback path. So for localhost it would look like "http://localhost:3000"
+10. Add a new [rule](https://manage.auth0.com/#/rules/create) in Auth0:
```
Comment (shakalee14) [inline README.md]: Can we add something more to ten to make it super clear? I used the 'empty' rule prompt, and then after it was successfully created, was able to test using: https://manage.auth0.com/#/rules and clicking on 'try all rules with ... (username-password-authentication)'. I'm not sure if it's worth adding testing instructions.

File: src/server/middleware/render-index.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const rollbarScript = process.env.ROLLBAR_CLIENT_TOKEN ?
 // good for doing dev offline
 const externalLinks = (process.env.NO_EXTERNAL_LINKS ? '' :
   `<link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Poppins">
-  <script src="https://cdn.auth0.com/js/lock/10.0.0-rc.2/lock.min.js"></script>`)
+  <script src="http://cdn.auth0.com/js/lock/11.0.1/lock.min.js"></script>`)
```
Comment (shakalee14) [inline src/server/middleware/render-index.js]: why is this served over http?

**Review by shakalee14 — APPROVED** [review]

**Comment by shakalee14** [comment]
I added a couple comments but this worked for me when I did the following:
 - cloned the project fresh, created new auth0 acct 
- followed set up instructions in your readme and added a new rule (using empty rule template)
- unset SUPPRESS_INVITE variable - i was able to create an org

**Comment by shakalee14** [comment]
Actually, I'm seeing in the logs this error:
`Error: Resolve function for "User.id" returned undefined`
and then
```ReqlRuntimeError: Cannot convert `undefined` with r.expr() in:
9:39:52 AM server.1               |  r.table("user_organization").getAll(undefined, {
9:39:52 AM server.1               |                                      ^^^^^^^^^  ^
9:39:52 AM server.1               |      index: "user_id"
9:39:52 AM server.1               |      ^^^^^^^^^^^^^^^^
9:39:52 AM server.1               |  }).filter({
9:39:52 AM server.1               |  ^
9:39:52 AM server.1               |      role: "ADMIN"
9:39:52 AM server.1               |  }).eqJoin("organization_id", r.table("organization"))("right").distinct()```

**Review by shakalee14 — CHANGES_REQUESTED** [review]
I think this branch works when a user is already authenticated and on admin sign up but doesn't work when I login (without an invite) and doesn't allow me to logout. When it does "work" I get the User.id undefined error. And when it doesn't work I get a `TypeError: Cannot read property 'given_name' of undefined`

File: src/server/index.js [hunk 3]
```diff
@@ -167,11 +167,11 @@ app.get('/login-callback',
       await User.save({
         auth0_id: req.user.id,
         // eslint-disable-next-line no-underscore-dangle
-        first_name: req.user._json.user_metadata.given_name,
+        first_name: req.user._json['https://spoke/user_metadata']['given_name'],
```
Comment (schuyler1d) [inline src/server/index.js]: will this break existing uses?  i.e. should we do something like:
 `const userMetadata = (req.user._json.user_metadata || req.user._json['https://spoke/user_metadata'] || {})`
... and then `userMetadata.given_name` ....

**Comment by shakalee14** [comment]
With passport-auth0 @ 0.5.2, and the added rule I can successfully logout.
I cannot login or signup still. If this does work for you, please let me know your config for passport auth0 and what context it works in. For us to merge, we need to make sure this branch works for the following use cases:
- already logged in user
- already signed up user, logging in
- logging out
- new user who is a texter (non admin)
- new user who is an admin 

I'm getting the following two errors:

For a new user:
```
10:51:38 AM server.1              |  { error: null value in column "cell" violates not-null constraint
10:51:38 AM server.1              |      at Connection.parseE (/Users/shaka/moveon/Spoke/node_modules/pg/lib/connection.js:567:11)
10:51:38 AM server.1              |      at Connection.parseMessage (/Users/shaka/moveon/Spoke/node_modules/pg/lib/connection.js:391:17)
10:51:38 AM server.1              |      at Socket.<anonymous> (/Users/shaka/moveon/Spoke/node_modules/pg/lib/connection.js:129:22)
10:51:38 AM server.1              |      at emitOne (events.js:96:13)
10:51:38 AM server.1              |      at Socket.emit (events.js:188:7)
10:51:38 AM server.1              |      at readableAddChunk (_stream_readable.js:172:18)
10:51:38 AM server.1              |      at Socket.Readable.push (_stream_readable.js:130:10)
10:51:38 AM server.1              |      at TCP.onread (net.js:542:20)
10:51:38 AM server.1              |  From previous event:
10:51:38 AM server.1              |      at Client_PG._query (/Users/shaka/moveon/Spoke/node_modules/knex/lib/dialects/postgres/index.js:239:12)
10:51:38 AM server.1              |      at Client_PG.query (/Users/shaka/moveon/Spoke/node_modules/knex/lib/client.js:197:17)
10:51:38 AM server.1              |      at Runner.<anonymous> (/Users/shaka/moveon/Spoke/node_modules/knex/lib/runner.js:146:36)
10:51:38 AM server.1              |  From previous event:
10:51:38 AM server.1              |      at /Users/shaka/moveon/Spoke/node_modules/knex/lib/runner.js:65:21
10:51:38 AM server.1              |      at tryOnImmediate (timers.js:543:15)
10:51:38 AM server.1              |      at processImmediate [as _immediateCallback] (timers.js:523:5)
10:51:38 AM server.1              |  From previous event:
10:51:38 AM server.1              |      at Runner.run (/Users/shaka/moveon/Spoke/node_modules/knex/lib/runner.js:51:31)
10:51:38 AM server.1              |      at Builder.Target.then (/Users/shaka/moveon/Spoke/node_modules/knex/lib/interface.js:35:43)
10:51:38 AM server.1              |      at insertFunc (/Users/shaka/moveon/Spoke/node_modules/rethink-knex-adapter/models.js:148:12)
10:51:38 AM server.1              |      at Object.dbModel.save (/Users/shaka/moveon/Spoke/node_modules/rethink-knex-adapter/models.js:179:16)
10:51:38 AM server.1              |      at _callee6$ (/Users/shaka/moveon/Spoke/src/server/index.js:186:18)
10:51:38 AM server.1              |      at tryCatch (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtime/runtime.js:65:40)
10:51:38 AM server.1              |      at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtime/runtime.js:303:22)
10:51:38 AM server.1              |      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtime/runtime.js:117:21)
10:51:38 AM server.1              |      at step (/Users/shaka/moveon/Spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
10:51:38 AM server.1              |      at /Users/shaka/moveon/Spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
10:51:38 AM server.1              |      at run (/Users/shaka/moveon/Spoke/node_modules/core-js/modules/es6.promise.js:66:22)
10:51:38 AM server.1              |      at /Users/shaka/moveon/Spoke/node_modules/core-js/modules/es6.promise.js:79:30
10:51:38 AM server.1              |      at flush (/Users/shaka/moveon/Spoke/node_modules/core-js/modules/_microtask.js:18:9)
10:51:38 AM server.1              |      at _combinedTickCallback (internal/process/next_tick.js:67:7)
10:51:38 AM server.1              |      at process._tickDomainCallback (internal/process/next_tick.js:122:9)
10:51:38 AM server.1              |    name: 'error',
10:51:38 AM server.1              |    length: 275,
10:51:38 AM server.1              |    severity: 'ERROR',
10:51:38 AM server.1              |    code: '23502',
10:51:38 AM server.1              |    detail: 'Failing row contains (8, auth0|5a68ab885481302d50b83ef9, , , null, newtestemail@gmail.com, 2018-01-24 10:51:38.294729-05, null, f, f).',
10:51:38 AM server.1              |    hint: undefined,
10:51:38 AM server.1              |    position: undefined,
10:51:38 AM server.1              |    internalPosition: undefined,
10:51:38 AM server.1              |    internalQuery: undefined,
10:51:38 AM server.1              |    where: undefined,
10:51:38 AM server.1              |    schema: 'public',
10:51:38 AM server.1              |    table: 'user',
10:51:38 AM server.1              |    column: 'cell',
10:51:38 AM server.1              |    dataType: undefined,
10:51:38 AM server.1              |    constraint: undefined,
10:51:38 AM server.1              |    file: 'execMain.c',
10:51:38 AM server.1              |    line: '1734',
10:51:38 AM server.1              |    routine: 'ExecConstraints' }
```

For old user:
After logging in at localhost:3000/login,
I am redirected to this page which gives a 'Not Found' error in the client: http://localhost:3000/-tR05U2epvzu4_DaRVCU7lbABDpwipBs

**Pull request merged**